content/asset-tracking/using-the-sdks.textile

```diff
@@ -0,0 +1,232 @@
+* @MAPBOX_ACCESS_TOKEN@ - Mapbox credentials
+* @GOOGLE_MAPS_API_KEY@ - Google Maps credentials
+
+You can set these values in your @~/.gradle/gradle.properties@ file.
+
+h2(#installing-sdk). Installing the SDK
+
+You can find information on installing the SDKs in the following resources:
+
+* "Instructions for Android":https://github.com/ably/ably-asset-tracking-android
+* "Instructions for iOS":https://github.com/ably/ably-asset-tracking-cocoa
+* "Instructions for JavaScript":https://github.com/ably/ably-asset-tracking-js
+
+h2(#publisher-sdk). Using the publisher SDK
+
+Common operations you will need to carry out on the publisher include:
+
+* Initialization of the publisher.
+* Start tracking an asset.
+* Stop tracking an asset.
+* Set the resolution constraints on an asset.
+
+h3(#publisher-initializing). Initializing the Publisher
+
+During initialization of the publisher various methods can be called to configure the @Publisher@.
+
+The required methods are:
+
+- connection := Called to provide Ably connection information, such as API keys, and any other configuration parameters as needed. 
+- map := Called to provide Mapbox configuration, such as API keys, any other configuration parameters as needed. 
+- resolutionPolicy := Sets the policy factory to be used to define the target resolution for publishers created from this builder.
+
+The optional methods are:
+
+- profile := Called to set the means of transport being used for the initial state of publishers created from this builder. If not set then the default value is @RoutingProfile.DRIVING@.
+- context := Called to provide the Android runtime context (on Android only).
+
+The following code example creates some example "resolution constraints":/asset-tracking#resolution-constraints:
+
+```[android] 
+// Prepare Resolution Constraints for an asset that will be used in the Resolution Policy
+val exampleConstraints = DefaultResolutionConstraints(
+    DefaultResolutionSet( // this constructor provides one Resolution for all states
+        Resolution(
+            accuracy = Accuracy.BALANCED,
+            desiredInterval = 1000L,
+            minimumDisplacement = 1.0
+        )
+    ),
+    proximityThreshold = DefaultProximity(spatial = 1.0),
+    batteryLevelThreshold = 10.0f,
+    lowBatteryMultiplier = 2.0f
+)
+```
+
+The next step is create a default "resolution":/asset-tracking#resolution to be used:
+
+```[android]
+// Prepare the default resolution for the Resolution Policy
+val defaultResolution = Resolution(Accuracy.BALANCED, desiredInterval = 1000L, minimumDisplacement = 1.0)
+```
+
+Once these are created you can then initialize the publisher with the constraints and default resolution, and start the publisher:
+
+```[android]
+// Initialise and Start the Publisher
+val publisher = Publisher.publishers() // get the Publisher builder in default state
+    .connection(ConnectionConfiguration(ABLY_API_KEY, CLIENT_ID)) // provide Ably configuration with credentials
```

Comment:
After landing of the [Token Auth PR]( https://github.com/ably/ably-asset-tracking-android/pull/222 ) this line will be obsolete

Reply:
@KacperKluka - cool, if possible can you let me know what this will be replaced with and I will update accordingly. Thanks.

Reply:
After digging around in the PR I've replaced the above code with what I think is going to be the new code:

```
// Initialise and Start the Subscriber
val subscriber = Subscriber.subscribers() // Get an AssetSubscriber
    .connection(
                ConnectionConfiguration(
                    Authentication.basic(
                        CLIENT_ID,
                        ABLY_API_KEY,
                    )
                )
            ) // provide Ably configuration with credentials
    .resolution( // request a specific resolution to be considered by the publisher
        Resolution(Accuracy.MAXIMUM, desiredInterval = 1000L, minimumDisplacement = 1.0)
    )
    .trackingId(trackingId) // provide the tracking identifier for the asset that needs to be tracked
    .start() // start listening for updates
```

Reply:
Right, we should replace 
``` kotlin
.connection(ConnectionConfiguration(ABLY_API_KEY, CLIENT_ID))
```
with
``` kotlin
.connection(ConnectionConfiguration(Authentication.basic(ABLY_API_KEY, CLIENT_ID)))
```

Reply:
Interesting, in the code I looked at last night API_KEY and CLIENT_ID were the other way around. I assume your comment is correct @KacperKluka ?

Reply:
Good catch, sorry for not paying enough attention to the order of the params, I've focused on adding the `Authentication.basic` and overlooked that the params are now passed in a different order :sweat_smile: So the correct version is
``` kotlin
.connection(ConnectionConfiguration(Authentication.basic(CLIENT_ID, ABLY_API_KEY)))
```

Reply:
Thanks @KacperKluka !